app/controllers/admin/deals_controller.rb

```diff
@@ -0,0 +1,55 @@
+module Admin
+  class DealsController < AdminController
+    before_action :set_deal, only: [:show, :edit, :update, :destroy]
+
+    def index
+      @deals = Deal.with_attached_images.all
+    end
+
+    def new
+      @deal = Deal.new
+    end
+
+    def create
+      @deal = Deal.new(deal_params)
+      respond_to do |format|
+        if @deal.save
+          format.html { redirect_to admin_deal_path(@deal), notice: "Deal created sucessfully"}
+        else
+          format.html { render :new }
+        end
+      end
+    end
+
+    def edit
+    end
+
+    def update
+      respond_to do |format|
+        if @deal.update(deal_params)
+          format.html { redirect_to admin_deal_path(@deal), notice: "Deal updated sucessfully" }
+        else
+          format.html { render :edit }
+        end
+      end
+    end
+
+    def destroy
+      @deal.destroy
+      respond_to do |format|
+        format.html { redirect_to admin_deals_path, alert: @deal.errors[:base][0] }
+        flash[:notice] = "Deal destroyed sucessfully"
```

Comment:
Lets start using I18n from now onwards. Discuss once with Akshay how we'll manage translation files. I got it configured it in his project 2 days back.